### DISCUSSION TITLE
XY-WFUSB

### DISCUSSION THREAD

**Discussion by borissov** [RavenSystem/esp-homekit-devices#2317] [MEPLHAA Scripts]
Hello, I am trying to get the device running with this configuration, but unfortunately, the button is not working, and I can't figure out where I am going wrong. 
```json
{
  "c": {
    "io": [
      [
        [
          5,
          14
        ],
        2
      ],
      [
        [
          4
        ],
        6
      ]
    ],
    "m": 60,
    "l": 14,
    "b": [
      [
        4,
        5
      ]
    ]
  },
  "a": [
    {
      "0": {
        "r": [
          [
            5
          ]
        ]
      },
      "1": {
        "r": [
          [
            5,
            1
          ]
        ]
      },
      "b": [
        [
          4
        ]
      ]
    }
  ]
}
```
Additional info for device:

https://templates.blakadder.com/sinilink_XY-WFUSB.html
https://community.home-assistant.io/t/sinilink-wifi-usb-controller-xy-wfusb-turns-usb-port-power-on-off-esphome-yaml/574532

Thanks!

**Comment by RavenSystem** [comment]
Enable internal pull-up resistor for GPIO 4 into "io" array: `[[4],6,1]`

**Reply by borissov** [reply]
Work perfect, thank you! 🙏
